### DISCUSSION TITLE
How to specify the custom path to zapp.yml/zapp.json in User settings.

### DISCUSSION THREAD

**Discussion by ShaswatChampatiray** [IBM/zopeneditor-about#338] [General]
We were able to resolve the copybooks in COBOL program by updating the zapp.yml in the below file structure.

workspace
└───.code-workspace 
└───COBOL
│   └───COPYBOOK
│   │   │   COPY.cpy
│   │   │   ...
│   │
│   └───SAM
│       │   SAM1.cbl
│       │   ...
└───.zapp.yaml

In the above file structure, the code-workspace file contains the zapp.yml reference path like mentioned below 

{
	"folders": [
		{
			"path": "../../../_dev/git-repos/custom-app/zopeneditor-sample"
		}
	],
	"settings": {}
}

We see in the workspace setting if we specify the folder attribute, open z editor by defaults picks it up from that location.

For the above scenario to work we have to create .code-workspace file for vscode, which happens when we create a workspace. 

**We want to know if the user doesnt create a workspace and still want to specify the reference path for zapp.yml, i.e. by using user settings (setting.json),**

**Comment by phaumer** [comment]
Hello @ShaswatChampatiray.
zapp.yaml files need to the in the top-level of a workspace. This is true for single root workspace as well as multi-root workspaces. For multi-root each workspace can have a zapp file and the property groups of each are combined into one list in the order you list the workspaces in the .workspaces file. Also see this [doc page](https://ibm.github.io/zopeneditor-about/Docs/setting_preferences.html) for more details.

From what I can see if your ascii graph it looks like that your zapp file is inside the COBOL directory? If you move it up to the top-level then in would be hopefully read. If you switch the [Logger to Debug mode](https://ibm.github.io/zopeneditor-about/Docs/locating_local_client_logs.html) then the output will also tell you which zapp files it is finding and reading at startup to troubleshoot the problem.
